interfaces/builtin/upower_observe.go

```diff
@@ -20,7 +20,12 @@
 package builtin
 
 import (
+	"bytes"
+	"fmt"
+
 	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
 )
 
 const upowerObserveConnectedPlugAppArmor = `
```

Comment:
@morphis - I thought that this was likely the case but the current implementation isn't enforcing the invariant that every interface should work standalone and implement everything required (ie, there is no concept of interface dependencies). In other words, if you are splitting these, it is perfectly reasonable in terms of snapd for developers to only implement one, or to implement each with two different snaps, or to implement both in one snap. With location-* it was decided that the small bit of PermamentSlot policy overlap wasn't anything to worry about, but you could, if you wanted, abstract some of this out so it is shared (not required by this PR).